### ISSUE TITLE
8.3.0-RC.3: adding a sparse index blows up sandbox

### ISSUE THREAD

**Issue by filmaj** [architect/architect#1010]
**Describe the issue**
If you introduce a sparse index (an index whose partition or composite key involves properties that are not part of the table's base partition or composite key), Sandbox explodes. This is not an issue in 8.2.1.

**Steps to reproduce**
1. Set up an arc app with a sparse index, i.e.:

```
@app
init

@http
get /

@tables
data
  account *String
  id **String

@indexes
data
  id *String

data
  ticket *String
  id **String
```

2. Run arc sandbox
3. You should see an error like:

```
ValidationException: One or more parameter values were invalid: Some index key attributes are not defined in AttributeDefinitions. Keys: [ticket, id], AttributeDefinitions: [account, id, id]
```

**Expected behavior**
Sandbox runs successfully.

**Additional context**
Comparing the parameters passed into dynalite's `createTable` function between 8.2.1 and 8.3.0-RC.3 with the above `app.arc` file, this is what you'll see:

<img width="934" alt="Screen Shot 2020-11-21 at 19 23 07" src="https://user-images.githubusercontent.com/52645/99890700-16f2df00-2c30-11eb-9581-fd5bb8cb941f.png">

The attribute definitions in 8.2.1 include one for each unique property name referenced from the sum total of a table definition + its indexes (in this case, one for `id`, `account` and `ticket`). In 8.3.0-RC.3, instead we see two entries for `id` and one for `account`. Seems like an entry for the `ticket` attribute is being overwritten or perhaps incorrectly written as `id`.

**Comment by filmaj** [comment]
For whatever reason, it seems like the inventory entry for that `ticket`/`id` index in my example above, as returned in https://github.com/architect/sandbox/blob/inventory/src/tables/create-table/_get-attribute-definitions.js#L10, looks like this:

```{
  name: 'dayta',
  partitionKey: 'id',
  partitionKeyType: 'String',
  sortKey: null,
  sortKeyType: null
}```

**Comment by filmaj** [comment]
OK one more hint, playing around with the `updates` branch of `@architect/inventory`, using the app.arc of my example above, it seems like retrieving the indexes using the `get` method returns only one index, whereas retrieving the indexes with via the properties returns both indexes:

```
➜ node
Welcome to Node.js v14.14.0.
Type ".help" for more information.
> inv=require('@architect/inventory')
[Function: architectInventory]
> inv().then((r) => { console.log(r.inv.indexes)})
Promise { <pending> }
> [
  {
    name: 'dayta',
    partitionKey: 'id',
    partitionKeyType: 'String',
    sortKey: null,
    sortKeyType: null
  },
  {
    name: 'dayta',
    partitionKey: 'ticket',
    partitionKeyType: 'String',
    sortKey: 'id',
    sortKeyType: 'String'
  }
]
> inv().then((r) => { console.log(r.get.indexes('dayta')) })
Promise { <pending> }
> {
  name: 'dayta',
  partitionKey: 'id',
  partitionKeyType: 'String',
  sortKey: null,
  sortKeyType: null
}
```

**Comment by filmaj** [comment]
I think it's the fact that for arrays, the `get` method in inventory uses `Array.find`, which returns only one item, but for indexes, there may be multiple matching a single table. https://github.com/architect/inventory/blob/updates/src/get.js#L9

**Comment by ryanblock** [comment]
Yeah sounds like that should be a filter!

**Comment by filmaj** [comment]
Changing it to a filter yields a malformed table definition in Sandbox:

```
create table params {
  TableName: 'init-staging-dayta',
  AttributeDefinitions: [
    { AttributeName: undefined, AttributeType: undefined },
    { AttributeName: 'id', AttributeType: 'S' },
    { AttributeName: 'ticket', AttributeType: 'S' },
    { AttributeName: 'id', AttributeType: 'S' }
  ],
  KeySchema: [
    { AttributeName: 'account', KeyType: 'HASH' },
    { AttributeName: 'id', KeyType: 'RANGE' }
  ],
  ProvisionedThroughput: { ReadCapacityUnits: 5, WriteCapacityUnits: 5 }
}
```

I will continue digging.

**Comment by ryanblock** [comment]
Resolved in Inventory 1.1.0 (specifically: https://github.com/architect/inventory/commit/72f5a62075684a2d95942594cf1ddd2ad43b1c34).
